lua/nvim-tree/diagnostics.lua

```diff
@@ -35,12 +34,25 @@ end
 local function from_nvim_lsp()
   local buffer_severity = {}
 
-  for buf, diagnostics in pairs(get_diagnostics()) do
-    if a.nvim_buf_is_valid(buf) then
-      local bufname = a.nvim_buf_get_name(buf)
-      if not buffer_severity[bufname] then
-        local severity = get_lowest_severity(diagnostics)
-        buffer_severity[bufname] = severity
+  if vim.diagnostic then
```

Comment:
Might be worth adding note here stating that this conditional should be removed when minimal required version of nvim is 0.7. Probably won't happen any time soon but it's a shame to leave dead code in.